### PR TITLE
Eris Upstream Merge Part 27: Nestor Remembers He Has A Server (Also FBPs and Synths are fully in)

### PR DESCRIPTION
**Pull request by drexample**
## About The Pull Request

A big fukken Eris upstream merge + FBP/Synth inclusion

## Why It's Good For The Game

just trust me bro

## Changelog
:cl:
tweak: UPSTREAM MERGE BAYBEEEEE
/:cl:

### CODE REVIEW

**Comment by EvilJackCarver** [comment]
I'm going to go ahead and flag the don't-merge on this until I finish reviewing it. This is a LOT of files to thumb through.

File: code/controllers/subsystems/tickets/tickets.dm [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+		return
+
+	var/datum/ticket/T = allTickets[N]
+	var/client/C = usr.client
+	if((T.staffAssigned && T.staffAssigned != C) || (T.lastStaffResponse && T.lastStaffResponse != C) || ((T.ticketState != TICKET_OPEN) && (T.ticketState != TICKET_STALE))) //if someone took this ticket, is it the same admin who is autoresponding? if so, then skip the warning
+		if(alert(usr, "[T.ticketState == TICKET_OPEN ? "Another admin appears to already be handling this." : "This ticket is already marked as closed or resolved"] Are you sure you want to continue?", "Confirmation", "Yes", "No") != "Yes")
+			return
+	T.assignStaff(C)
+
+	var/response_phrases = list("Thanks" = "Thanks, have a Paradise day!",
```
Comment (EvilJackCarver) [inline code/controllers/subsystems/tickets/tickets.dm:226]: Did they pull this from Paradise and not change the text?

File: code/controllers/subsystems/tickets/tickets.dm [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+		"Handling It" = "The issue is being looked into, thanks.",
+		"Already Resolved" = "The problem has been resolved already.",
+		//"Mentorhelp" = "Please redirect your question to Mentorhelp, as they are better experienced with these types of questions.",
+		"Happens Again" = "Thanks, let us know if it continues to happen.",
+		"Github Issue Report" = "To report a bug, please go to our <a href='[config.githuburl]'>Github page</a>. Then go to 'Issues'. Then 'New Issue'. Then fill out the report form. If the report would reveal current-round information, file it after the round ends.",
+		"Clear Cache" = "To fix a blank screen, go to the 'Special Verbs' tab and press 'Reload UI Resources'. If that fails, clear your BYOND cache (instructions provided with 'Reload UI Resources'). If that still fails, please adminhelp again, stating you have already done the following." ,
+		"IC Issue" = "This is an In Character (IC) issue and will not be handled by admins. You could speak to Security, Internal Affairs, a Departmental Head, Nanotrasen Representetive, or any other relevant authority currently on station.",
```
Comment (EvilJackCarver) [inline code/controllers/subsystems/tickets/tickets.dm:233]: Typo in "Representative"

File: code/game/jobs/department.dm [hunk 3]
```diff
@@ -107,7 +107,8 @@
 /datum/department/church
 	name = "Children of Mekhane"
 	id = DEPARTMENT_CHURCH
-	funding_type = FUNDING_EXTERNAL
+	account_budget = 4500 //each Neotheo has a wage of 900, this is enough to pay 5 paychecks before needing more cash
+	funding_type = FUNDING_NONE //The church on eris has no external funding. This further reinforces the theory that everyone on the CEV Eris is a reject of their factions
```
Comment (EvilJackCarver) [inline code/game/jobs/department.dm:111]: Are we sure we want to throw the Church's budget to the fire like this?

File: code/game/jobs/job/security.dm [hunk 4]
```diff
@@ -12,6 +12,7 @@
 	also_known_languages = list(LANGUAGE_GERMAN = 10)
 	req_admin_notify = 1
 	wage = WAGE_COMMAND
+	also_known_languages = list(LANGUAGE_CYRILLIC = 60, LANGUAGE_SERBIAN = 60, LANGUAGE_NEOHONGO = 100)
```
Comment (EvilJackCarver) [inline code/game/jobs/job/security.dm:15]: It'd be better to combine them into the first one or you'll lose the chance to know German.

**Review by EvilJackCarver — CHANGES_REQUESTED** [review]
Gonna pump the brakes on reviewing for now. Here's what I've turned up so far.

File: code/modules/projectiles/guns/projectile/automatic/ak47.dm [hunk 5]
```diff
@@ -77,3 +125,126 @@
...
+	icon = 'icons/obj/guns/projectile/ak/wasp.dmi'
+	random_sprite = FALSE
+	spawn_blacklisted = TRUE
+
+/obj/item/weapon/gun/projectile/automatic/ak47/fs/brush
+	name = "FS AR .30 \"Brushmaster\""						// A small play on Bushmaster
+	desc = "Weapon of the oppressed, oppressors, and extremists of all flavours. \
+			This is a copy of an ancient semi-automatic rifle chambered for .30 Rifle. If it won't fire, percussive maintenance should get it working again. \
+			It is known for its easy maintenance, and low price. This gun is not in active military service anymore, but has become ubiquitous among criminals and insurgents. \
+			This cheap copy looks as tacticool as possible. Emulating the IH PD rifle in most ways, even the the fixed skeleton stock looks the same."
```
Comment (EvilJackCarver) [inline code/modules/projectiles/guns/projectile/automatic/ak47.dm:188]: Ironhammer reference

File: code/modules/projectiles/guns/projectile/automatic/ak47.dm [hunk 6]
```diff
@@ -77,3 +125,126 @@
...
+
+
+//////////////////////////////////////////IH//////////////////////////////////////////
+
+/obj/item/weapon/gun/projectile/automatic/ak47/fs/ih
+	name = "FS AR .30 \"Venger\""						// From a song
+	desc = "Weapon of the oppressed, oppressors, and extremists of all flavours. \
+			This is a copy of an ancient semi-automatic rifle chambered for .30 Rifle. If it won't fire, percussive maintenance should get it working again. \
+			It is known for its easy maintenance, and low price. This gun is not in active military service anymore, but has become ubiquitous among criminals and insurgents. \
+			This rifle is from the Frozen Star's Planetary Defence line. Painted in IH blue and black, with a folding stock so it can be stored compactly for years without use."
```
Comment (EvilJackCarver) [inline code/modules/projectiles/guns/projectile/automatic/ak47.dm:201]: Ironhammer reference

File: code/modules/projectiles/guns/projectile/automatic/motherfucker.dm [hunk 7]
```diff
@@ -0,0 +1,45 @@
+/obj/item/weapon/gun/projectile/automatic/motherfucker
+	name = "HM Motherfucker .35 \"Punch Hole\""
+	desc = "A 6 barrel, pump action carbine, shakes like the devil. but will turn anything in a 90º from you in swiss cheese."
```
Comment (EvilJackCarver) [inline code/modules/projectiles/guns/projectile/automatic/motherfucker.dm:3]: Consider rewriting the final part of the description to ..."but will turn anything inside a 90° cone in front of you into Swiss cheese"

File: code/modules/projectiles/guns/projectile/boltgun.dm [hunk 8]
```diff
@@ -116,3 +118,25 @@
 	matter = list(MATERIAL_STEEL = 20, MATERIAL_WOOD = 10)
 	wielded_item_state = "_doble"
 	rarity_value = 48
+	spawn_blacklisted = FALSE
+
+/obj/item/weapon/gun/projectile/boltgun/handmade
+	name = "handmade bolt action rifle"
+	desc = "A handmade bolt action rifle, made from junk. and some spare parts."
```
Comment (EvilJackCarver) [inline code/modules/projectiles/guns/projectile/boltgun.dm:125]: ..."made from junk and some spare parts."

File: code/modules/projectiles/guns/projectile/revolver/handmade.dm [hunk 9]
```diff
@@ -0,0 +1,13 @@
+/obj/item/weapon/gun/projectile/revolver/handmade
+	name = "handmade revolver"
+	desc = "Handmade revolver, made from gun parts. and some duct tap, will it even hold up?"
```
Comment (EvilJackCarver) [inline code/modules/projectiles/guns/projectile/revolver/handmade.dm:3]: Consider rewriting to "A handmade revolver, made from gun parts and some duct tape. Will it even hold up?"

File: code/modules/projectiles/guns/projectile/sniper.dm [hunk 10]
```diff
@@ -1,9 +1,10 @@
 /obj/item/weapon/gun/projectile/heavysniper
 	name = "SA AMR \"Hristov\""
-	desc = "A portable anti-armour rifle, fitted with a scope, it was originally designed for use against armoured exosuits. It is capable of punching through windows and non-reinforced walls with ease. Fires armor piercing 14.5mm shells. Alt click it to access the chamber."
+	desc = "A portable anti-armour rifle, fitted with a night-vision scope, it was originally designed for use against armoured exosuits. It is capable of punching through windows and non-reinforced walls with ease, but suffers from overpenetration at close range. Fires armor piercing .60 shells. Can be upgraded using thermal glasses."
```
Comment (EvilJackCarver) [inline code/modules/projectiles/guns/projectile/sniper.dm:3]: UK-ENG variant spelling

**Review by EvilJackCarver — CHANGES_REQUESTED** [review]
Still not finished thumbing through the changes, but got a load more for ya.

File: code/modules/research/xenoarchaeology/tools/anomaly_suit.dm [hunk 11]
```diff
@@ -1,14 +1,53 @@
...
+/obj/item/clothing/head/space/void/hazardhelmet/moebius
+	name = "moebius hazard helmet"
+	desc = "A moebius branded violet voidsuit helmet with a large visor that hides your face. Integrated heads-up display not included."
+	icon_state = "hev_violet_helmet"
+	item_state = "hev_violet_helmet"
+	spawn_blacklisted = TRUE
+
+/obj/item/clothing/suit/space/void/hazardsuit/moebius
+	name = "moebius hazard voidsuit"
+	desc = "A moebius branded violet voidsuit that is capable of protecting its user against most hostile environment conditions, including anomalous particles. Integrated self-diagnostics system not included!"
```
Comment (EvilJackCarver) [inline code/modules/research/xenoarchaeology/tools/anomaly_suit.dm:49]: Moebius reference

File: maps/CEVEris/overmap-eris.dm [hunk 12]
```diff
@@ -6,6 +6,9 @@
 	speed_mod = 5 SECONDS
 	base = 1
 
+	name_stages = list("CEV Eris", "unknown vessel", "unknown spatial phenomenon")
```
Comment (EvilJackCarver) [inline maps/CEVEris/overmap-eris.dm:9]: CEV Eris mention

File: maps/submaps/planetary_ruins/ec_old_crash/ec_old_crash.dm [hunk 13]
```diff
@@ -87,11 +89,15 @@
 	Senior Explorer [S.get_random_name(pick(MALE,FEMALE))]<br>
 	Senior Explorer [S.get_random_name(pick(MALE,FEMALE))]<br>
 	Explorer [S.get_random_name(pick(MALE,FEMALE))]<br>
-	I am Lieutenant Hao Ru, captain of [shipname] of the Terran Commonwealth Expeditionary Corps. I will be joining my crew in cryo now.<br>
+	I am Lieutenant Hao Ru, captain of [shipname] of the Hansa Trade Union. I will be joining my crew in cryo now.<br>
```
Comment (EvilJackCarver) [inline maps/submaps/planetary_ruins/ec_old_crash/ec_old_crash.dm:92]: Metro 2033 reference, if we removed the one above

**Comment by EvilJackCarver** [comment]
As a note (and because I want to separate it from the review), the changes to the icon code make me feel that it should be tested before merge, especially with reference to the modular files.

**Review by EvilJackCarver — CHANGES_REQUESTED** [review]
If you're reading the cliffnotes version on Discord, I'd recommend you open it up in your browser. I've probably edited it a few times since it got posted by the bot.

The short version is - we do plan to merge this. We're just busy.

The long version? Well...

I've heard a few people wondering about the state of this PR. It's not in limbo - we *are* planning to merge it at some point soon. However -

* I made a request to Nestor that he not work on this before he finished up the dispatcher, as I really wasn't up for fixing a fourth merge conflict - Nestor has agreed to that (and it's much appreciated).
* I cannot finish up the dispatcher myself, as I lack the required knowledge, experience, and a suitable testing environment for the dispatcher's webhook code.
* Nestor has been busy in real life with real life things (like uni), which has been keeping him from being able to work on the dispatcher.
* I'm the one blocking this PR from being merged, as there are unresolved issues, plus I would at least like to get Nestor's opinion on whether we want to throw away the entirety of the Church's budget ([see here](https://github.com/Eclipse-Station/NEV-Northern-Light/pull/1479#discussion_r627072091)).
* I would make a majority of the fixes myself, but I'm dealing with a bad bout of depression myself at the moment and haven't had the time nor the energy to work on it. Plus, I'm a lazy mike-foxtrot outside of the depression bit, anyway.

This *will be* merged in due time, don't worry - we're just busy with other things at the moment.

**Comment by benj8560** [comment]
was testing the church stuff for this PR and honestly even tho I love the new holiness system it feels like it lacks much impact. I var edited myself to full and charge times where still rather useless. Perhaps would be better if oblisks had considerably faster charge boost and maybe even give  cruciforms a lot more capacity. Also plasable holiness is just not working as intended since I seen almost no difference from the start charge speed and upgraded. This was awhile ago so could be these issues already got addressed. I told some other players about it when I was testing it and they seemed equally concerned about how bad it was.

File: code/controllers/subsystems/tickets/tickets.dm (hunk 1 above)
Comment (drexample) [reply code/controllers/subsystems/tickets/tickets.dm:226]: yes

File: code/game/jobs/department.dm (hunk 3 above)
Comment (drexample) [reply code/game/jobs/department.dm:111]: We had intents on introducing more scarcity and ship is pretty deep away from any centralized banking, so let's see how that workks.

File: code/modules/admin/bluespacetech.dm [hunk 14]
```diff
@@ -292,7 +310,7 @@ ADMIN_VERB_ADD(/client/proc/cmd_dev_bst, R_ADMIN|R_DEBUG, TRUE)
 
 /obj/item/weapon/card/id/bst
 	icon_state = "centcom"
-	desc = "An ID straight from Central Command. This one looks highly classified."
+	desc = "An ID straight from Hansa. This one looks as though its very existence is a trade secret."
```
[the remark this answers is not in the document]
Comment (drexample) [reply code/modules/admin/bluespacetech.dm:313]: Hansa sector is not a Metro reference.

[the remark this answers is not in the document]
Comment (drexample) [reply code/modules/admin/bluespacetech.dm:313]: It's an eris lore "faction"

File: code/modules/clothing/suits/miscellaneous.dm [hunk 15]
```diff
@@ -419,3 +409,20 @@
 		bio = 0,
 		rad = 0
 	)
+
+/obj/item/clothing/suit/storage/scavengerarmor
+	name = "Scavenger armor"
+	desc = "A sturdy, rigged Scavenger armor. strong and sturdy as most vests. made fully from junk."
```
[the remark this answers is not in the document]
Comment (drexample) [reply code/modules/clothing/suits/miscellaneous.dm:415]: copypaste issue, probably

File: code/modules/long_range_scanner/long_range_scanner.dm [hunk 16]
```diff
@@ -0,0 +1,482 @@
...
+/obj/machinery/power/long_range_scanner/proc/log_event(var/event_type, var/atom/origin_atom)
+	var/logstring = "[stationtime2text()]: "
+	switch (event_type)
+
+		if (EVENT_ENABLED to EVENT_RECONFIGURED)
+			switch (event_type)
+				if (EVENT_ENABLED)
+					logstring += "Shield powered up"
+				if (EVENT_DISABLED)
+					logstring += "Shield powered down"
```
[the remark this answers is not in the document]
Comment (drexample) [reply code/modules/long_range_scanner/long_range_scanner.dm:338]: //This proc keeps an internal log of shield impacts, activations, deactivations, and a vague log of config changes

File: code/game/jobs/department.dm (hunk 3 above)
Comment (EvilJackCarver) [reply code/game/jobs/department.dm:111]: Alright, then.

File: code/game/jobs/job/security.dm (hunk 4 above)
Comment (EvilJackCarver) [reply code/game/jobs/job/security.dm]: There's four other entries in that file that suffer with the same problem, but I'll get 'em taken care of here in a few.

**Review by EvilJackCarver — APPROVED** [review]
Everything looks OK from my end. Just make sure the checks pass before you merge it.

I'll be releasing the DNM flag momentarily.

**Pull request merged**